### ISSUE TITLE
Extensions on general API Info

### ISSUE THREAD

**Issue by ymohl-cl** [swaggo/swag#398]
**Is your feature request related to a problem? Please describe.**
Hello ;)
I need implement on the General API Info the extensions fields like this x-...
In my case, it's to x-google-endpoints but others cases exist.
It's mandatory in my case to the deployment in google cloud platform and extensions should be on the root level in the swagger_file.yaml/json

extension on OpenAPI: https://swagger.io/docs/specification/openapi-extensions/

**Describe the solution you'd like**
Add a parsing from the root level in the function ParseGeneralAPIInfo with "@x-" to the pattern detection of x-name-extension key.
To an easy implementation, i suggest define the value in json format on a line like this example:

```
// @x-google-endpoints [{"name":"name.endpoints.environment.cloud.goog","allowCors":true}]
```

output yaml:
```
x-google-endpoints:
- allowCors: true
  name: name.endpoints.environment.cloud.goog
```

output json:
```
    "x-google-endpoints": [
        {
            "allowCors": true,
            "name": "name.endpoints.environment.cloud.goog"
        }
    ],
```

The dev is did for my usage and i like suggest you on a pull request now if you agree ;)
Thanks a lot !

Great day !

**Comment by Nerzal** [comment]
If you have allready implemented it, i guess you can just create a PR

**Comment by ymohl-cl** [comment]
ty, it's done ;)

**Comment by easonlin404** [comment]
fixes via #416
